New Bank/NewBank/newbank/server/NewBank.java

```diff
@@ -82,4 +107,17 @@ private String moveFunds(CustomerID customer,double amount,String from,String to
 		}
 		return "FAIL";
 	}
+
+	private String payment(Customer customer, double amount,String from,String to,  Customer custCredit, String debitBankName, String creditBankName ) {
+		if(amount > 0) { // should never need to move a negative amount
+			if(customer.editAccountBalance(from,-amount) && customer.getBankName().equals(debitBankName)) { // try to remove amount
+				if(custCredit.getBankName().equals(creditBankName)&&custCredit.editAccountBalance(to,amount)) { // try to add amount
```

Comment:
I think that the bankName is not needed in this method. It seems to only be used to check if the bankName is correct, which it can only do because it has already correctly identified the customer which is the part that is used for the payment.

If you remove the '&& customer.getBankName().equals(debitBankName)' in the first if condition and the '&& customer.getBankName().equals(debitBankName)' in the second if condition and then run the program, you can still make the payments regardless of what you enter as the bank name. So it seems that the payment method works as intended without needing the bankName.

Also, as this seems to be the only use for bankName, the bankName variable can probably be removed from the rest of the code as I don't think it's introducing anything that isn't already captured by the Customer and CustomerID classes.

Reply:
I agree may be unnecessary but I would prefer to keep this as it reflects what we see in reality where we do need to sometimes enter a bank name